pyccel/parser/scope.py

```diff
@@ -734,8 +798,13 @@ def get_new_name(self, current_name = None, is_temp = None):
             # When a name is suggested, try to stick to it
             new_name,_ = create_incremented_string(self.all_used_symbols, prefix = current_name)
 
-        new_name = PyccelSymbol(new_name, is_temp = is_temp)
-        return self.insert_symbol(new_name)
+        collisionless_symbol = self.name_clash_checker.get_collisionless_name(new_name,
```

Comment:
Could we call this variable `collisionless_name`?

Reply:
Done : 80879d4cf